src/frontend/src/tests/lib/derived/tokens.derived.spec.ts

```diff
@@ -0,0 +1,110 @@
+import * as btcEnv from '$env/networks.btc.env';
+import * as ethEnv from '$env/networks.eth.env';
+import { BTC_MAINNET_TOKEN, BTC_TESTNET_TOKEN } from '$env/tokens.btc.env';
+import { ETHEREUM_TOKEN, ICP_TOKEN, SEPOLIA_TOKEN } from '$env/tokens.env';
+import { erc20DefaultTokensStore } from '$eth/stores/erc20-default-tokens.store';
+import { erc20UserTokensStore } from '$eth/stores/erc20-user-tokens.store';
+import type { Erc20Token } from '$eth/types/erc20';
+import type { Erc20UserToken } from '$eth/types/erc20-user-token';
+import { icrcCustomTokensStore } from '$icp/stores/icrc-custom-tokens.store';
+import { icrcDefaultTokensStore } from '$icp/stores/icrc-default-tokens.store';
+import type { IcToken } from '$icp/types/ic-token';
+import type { IcrcCustomToken } from '$icp/types/icrc-custom-token';
+import * as appContants from '$lib/constants/app.constants';
+import { tokens } from '$lib/derived/tokens.derived';
+import { testnetsStore } from '$lib/stores/settings.store';
+import { parseTokenId } from '$lib/validation/token.validation';
+import { mockValidErc20Token } from '$tests/mocks/erc20-tokens.mock';
+import { mockValidIcToken } from '$tests/mocks/ic-tokens.mock';
+import { get } from 'svelte/store';
+import { expect } from 'vitest';
+
+describe('tokens.derived', () => {
+	const mockErc20DefaultToken: Erc20Token = {
+		...mockValidErc20Token,
+		id: parseTokenId('Erc20DefaultTokenId'),
+		symbol: 'DTK',
+		address: `${mockValidErc20Token.address}1`
+	};
+
+	const mockEr20UserToken: Erc20UserToken = {
+		...mockValidErc20Token,
+		id: parseTokenId('Erc20UserTokenId'),
+		symbol: 'EUTK',
+		address: `${mockValidErc20Token.address}2`,
+		version: undefined,
+		enabled: true
+	};
+
+	const mockIcrcDefaultToken: IcToken = {
+		...mockValidIcToken,
+		ledgerCanisterId: `${mockValidIcToken.ledgerCanisterId}1`
+	};
+
+	const mockIcrcCustomToken: IcrcCustomToken = {
+		...mockValidIcToken,
+		ledgerCanisterId: `${mockValidIcToken.ledgerCanisterId}2`,
+		version: 1n,
+		enabled: true
+	};
+
+	describe('tokens', () => {
+		beforeEach(() => {
+			vi.resetAllMocks();
+
+			erc20DefaultTokensStore.reset();
```

Comment:
In the Gix channel d.K. recently shared a snippet to reset all stores of the app for test purpose if I remember correctly. Maybe it's something we can reuse? (for another PR)